src/components/DocumentationTopic/TopicsLinkCardGridItem.vue

```diff
@@ -0,0 +1,93 @@
+<!--
+  This source file is part of the Swift.org open source project
+
+  Copyright (c) 2022 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+<template>
+  <Card
+    class="reference-card-grid-item"
+    :url="item.url"
+    :image="imageReferences.card"
+    :title="item.title"
+    floating-style
+    :size="compact ? undefined: 'large'"
+    :link-text="compact? '': linkText"
+  >
+    <template #cover="{ classes }" v-if="!imageReferences.card">
+      <div :class="classes" class="reference-card-grid-item__image">
+        <TopicTypeIcon
+          :type="item.role"
+          :image-override="references[imageReferences.icon]"
+          class="reference-card-grid-item__icon"
+        />
+      </div>
+    </template>
+    <ContentNode v-if="!compact" :content="item.abstract" />
+  </Card>
+</template>
+
+<script>
+import Card from 'docc-render/components/Card.vue';
+import TopicTypeIcon from 'docc-render/components/TopicTypeIcon.vue';
+import { TopicRole } from 'docc-render/constants/roles';
+
+export const ROLE_LINK_TEXT = {
```

Comment:
Nothing to change here yet, but we may want to keep note of these new hardcoded strings for future localization purposes.